api/tests/python/end_points/test_namex_search.py

```diff
@@ -346,3 +359,204 @@ def test_namex_search_last_name(client, jwt, app, search_name):
+    rv = client.get(
+        f'api/v1/requests?submittedStartDateTime={submitted_start_date_time}&rows=1000',
+        headers=create_header(jwt, [User.VIEWONLY])
+    )
+    data = rv.data
+    assert data
+    resp = json.loads(data.decode('utf-8'))
+
+    assert resp.get('nameRequests') and resp.get('response')
+    response_count = len(resp['nameRequests'][0])
+    assert response_count >= 0
+    assert response_count == expected_result_count
+
+
+@pytest.mark.parametrize('submitted_end_date_time, expected_result_count', [
+    (get_utc_now_with_max_delta_str(timedelta(days=-1)), 8),
+    (get_utc_now_with_max_delta_str(timedelta(days=-32)), 3),
+    (get_utc_now_with_max_delta_str(timedelta(-5*(365))), 1),
+    (get_utc_now_with_min_delta_str(timedelta(-5*(365))), 0),
+    (get_utc_now_with_max_delta_str(timedelta(-6*(365))), 0),
+    (get_utc_now_min_time_str(), 8),
+    (get_utc_now_max_time_str(), 10),
+    (get_utc_now_with_max_delta_str(timedelta(days=1)), 11),
+    (get_utc_now_with_max_delta_str(timedelta(days=100)), 13),
+    (get_utc_now_with_max_delta_str(timedelta(days=5*(365)+1)), 15),
+    (get_utc_now_with_max_delta_str(timedelta(days=6*(365))), 15),
+])
+def test_namex_search_submitted_end_date_time(client, jwt, app, submitted_end_date_time, expected_result_count):
+    """Test searching by submitted end date."""
+
+    submitted = [
+        get_utc_now_with_delta(timedelta(days=-5*(365))),
+        get_utc_now_with_delta(timedelta(days=-1*(365))),
+        get_utc_now_with_delta(timedelta(days=-100)),
+        get_utc_now_with_delta(timedelta(days=-31)),
+        get_utc_now_with_delta(timedelta(days=-15)),
+        get_utc_now_with_delta(timedelta(days=-1)),
+        get_utc_now_with_delta(timedelta(days=-1)),
+        get_utc_now_with_delta(timedelta(days=-1)),
+        get_utc_now(),
+        get_utc_now(),
+        get_utc_now_with_delta(timedelta(days=1)),
+        get_utc_now_with_delta(timedelta(days=31)),
+        get_utc_now_with_delta(timedelta(days=100)),
+        get_utc_now_with_delta(timedelta(days=1*(365))),
+        get_utc_now_with_delta(timedelta(days=5*(365))),
+    ]
+    generate_nrs(len(submitted), [], [], submitted)
+
+    # get the resource (this is what we are testing)
+    rv = client.get(
+        f'api/v1/requests?submittedEndDateTime={submitted_end_date_time}&rows=1000',
+        headers=create_header(jwt, [User.VIEWONLY])
+    )
+    data = rv.data
+    assert data
+    resp = json.loads(data.decode('utf-8'))
+
+    assert resp.get('nameRequests') and resp.get('response')
+    response_count = len(resp['nameRequests'][0])
+    assert response_count >= 0
+    assert response_count == expected_result_count
+
+# todo add test for submitted start and end date time
+
+@pytest.mark.parametrize('submitted_start_date_time, submitted_end_date_time', [
+    (escape_date_time('2021-05-11 00:00:00-07:00'), escape_date_time('2021-05-01 23:59:59-07:00')),
```

Comment:
These date-time formats are inconsistent with our other UIs (eg, 2021-01-01T00:00:00.000000+00:00).

Also they are somewhat confusing ... if I understand correctly, the first example is "May 11, 2021 at 12:00:00 am in a timezone that is 7h behind UTC". But normally, date-times passed across the network are just plain UTC.

A couple of us chatted about this, this morning... I will let Cam discuss this with you further.

Reply:
yes, talked to Cam a bit and will be refactoring such that FE won't need to provide anything but the date.  We discuss more after things are refactored.